### DISCUSSION TITLE
BUG: Commits on target branch showing on a PR

### DISCUSSION THREAD

**Discussion by dzenilee** [community/community#61091] [Pull Requests]
### Select Topic Area

Bug

### Body

This bug still persists: https://stackoverflow.com/questions/16306012/github-pull-request-showing-commits-that-are-already-in-target-branch

In other words, commits already on the _target_ branch are unnecessarily showing up in the commit history of a PR, not just the new commits I want to commit to the target branch. 

What I did: 
```
# from the feature branch 

git rebase main
git pull --rebase
git push
```

This workaround worked at some point, but it doesn't anymore: https://stackoverflow.com/a/46782679/7554627

**Comment by redelschaap** [comment]
Yes, it's very annoying. Would love to see them fix this strange behavior.

**Comment by dzenilee** [comment]
I had the order reversed above. Try: 

```
git pull --rebase
git rebase main 
git push -f 

git log main..  # will tell you what commits will be included on your PR (first make sure main's up to date)
```

**Comment by claresloggett** [comment]
I have seen statements that this is by design, but I would agree that this is a bug. It's unexpected behaviour, and makes it impossible for the user to tell from the PR UI what changes will actually be applied when the merge is made. The PR diff ought to be updated automatically by GitHub whenever the source or target branches are pushed to.

**Comment by jhelbig** [comment]
This is most definitely a bug.  I believe the problem is that instead of comparing between two branches in their active state, they are now comparing between two commits.

I have no hard evidence this is what is happening, but I certainly get the same results in the PR when I diff two commits across branches.

This is a problem for me when performing reviews on long-running project branches with multiple smaller feature branches.  The amount of commits and changes that need to be "reviewed" continues to grow the further away I get from the project branch.  After three PRs I'm reviewing the same changes I reviewed in the first PR.  I still feel compelled to review them because IDK if there are **actual** changes or if they are these _phantom_ commits.

**Comment by nadalaba** [comment]
Unfortunately, it seems unlikely that GitHub will change this behavior any time soon, see #115266.

**Comment by bcowell-riis** [comment]
Still having this issue
